### PR TITLE
[FIX] point_of_sale,pos_restaurant: load data on new pos config

### PR DESCRIPTION
**Pull request by loukwets**
In this commit 2 issues are fixed. The first issue is when trying to load demo data when the default pos configs are deleted. The second is when trying to reload onboarding data when the products are already loaded but archived (task-3539455).

steps to reproduce the first issue:
- load db without demo data
- delete pos created by default
- create a new pos config
- open and try to load demo data

steps to reproduce the second issue:
- load db without demo data
- open pos and load demo data
- archive products
- open pos and try to load demo data

changes:
- main pos config demo data has been separated from onboarding data.
- condition has been added in the search count of _pos_has_valid_product method to check for archived products.

Enterprise PR: https://github.com/odoo/enterprise/pull/48621

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -2194,24 +2194,27 @@ def log_partner_message(self, partner_id, action, message_type):
...
         config = self.env.ref('point_of_sale.pos_config_main', raise_if_not_found=False)
-        config.with_context(bypass_categories_forbidden_change=True).write({
-            'limit_categories': True,
-            'iface_available_categ_ids': [Command.link(self.env.ref('point_of_sale.pos_category_miscellaneous').id), Command.link(self.env.ref('point_of_sale.pos_category_desks').id), Command.link(self.env.ref('point_of_sale.pos_category_chairs').id)]
-        })
+        if config:
+            config.with_context(bypass_categories_forbidden_change=True).write({
+                'limit_categories': True,
+                'iface_available_categ_ids': [Command.link(self.env.ref('point_of_sale.pos_category_miscellaneous').id), Command.link(self.env.ref('point_of_sale.pos_category_desks').id), Command.link(self.env.ref('point_of_sale.pos_category_chairs').id)]
+            })
```
Comment (caburj) [inline addons/point_of_sale/models/pos_session.py:2217]: I think there is no need for this change. But test before reverting it.

File: addons/pos_restaurant/models/pos_session.py [hunk 2]
```diff
@@ -67,21 +67,24 @@ def get_onboarding_data(self):
...
-        configs.with_context(bypass_categories_forbidden_change=True).write({
-            'limit_categories': True,
-            'iface_available_categ_ids': [Command.link(self.env.ref('pos_restaurant.food').id), Command.link(self.env.ref('pos_restaurant.drinks').id)]
-        })
+        configs = self.config_id.filtered('module_pos_restaurant')
+        if configs:
+            configs.with_context(bypass_categories_forbidden_change=True).write({
+                'limit_categories': True,
+                'iface_available_categ_ids': [Command.link(self.env.ref('pos_restaurant.food').id), Command.link(self.env.ref('pos_restaurant.drinks').id)]
+            })
```
Comment (caburj) [inline addons/pos_restaurant/models/pos_session.py:87]: the same comment above https://github.com/odoo/odoo/pull/138035/files#r1350367936

**Comment by caburj** [comment]
@odoo/rd-security Hi team. We triggered the ci/security in this PR (and its linked PR in enterprise). This PR is about splitting the onboarding data into multiple files so that when certain conditions are not met, the portion of the onboarding data won't be loaded to prevent crashes. Because of the split, we need to call `convert.convert_file` in more places. Can we ask for override?

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: convert_file to static path

**Comment by caburj** [comment]
robodoo r+

**Pull request closed without merging**